cognite_toolkit/_cdf_tk/loaders/_resource_loaders/data_organization_loaders.py

```diff
@@ -94,16 +94,31 @@ def dump_id(cls, id: str) -> dict[str, Any]:
 
     def load_resource(self, resource: dict[str, Any], is_dry_run: bool = False) -> DataSetWrite:
         if resource.get("metadata"):
-            for key, value in resource["metadata"].items():
-                resource["metadata"][key] = json.dumps(value)
-        if resource.get("writeProtected") is None:
-            # Todo: Setting missing default value, bug in SDK.
-            resource["writeProtected"] = False
-        if resource.get("metadata") is None:
-            # Todo: Wrongly set to empty dict, bug in SDK.
-            resource["metadata"] = {}
+            for key, value in list(resource["metadata"].items()):
+                if isinstance(value, dict):
+                    resource["metadata"][key] = json.dumps(value)
```

Comment:
This is a bit surprising. You can use objects in the metadata that is automatically made into a json string.